### PR TITLE
Add telemetry pings (part 1 - UI instrumentation)

### PR DESCRIPTION
**Pull request by k88hudson**
This patch adds instrumentation for user event telemetry to AS router templates – nothing is actually done with the events so far, it just adds a simple ping action type.
The first commit is just some style and functionality changes to support actual links.

### CODE REVIEW

File: system-addon/content-src/message-center/components/SnippetBase.jsx [hunk 1]
```diff
@@ -6,11 +6,14 @@ const defaultStyles = {
     bottom: 0,
     left: 0,
     right: 0,
-    padding: "20px",
-    backgroundColor: "white",
+    padding: "12px",
+    backgroundColor: "var(--newtab-card-background-color)",
```
Comment (k88hudson) [inline system-addon/content-src/message-center/components/SnippetBase.jsx]: @rlr do you think this is a good option for the snippets background so it works with dark theme?

**Comment by ncloudioj** [comment]
r+ for the telemetry part. Thanks @k88hudson  for taking the first crack at it, it looks great!

@rlr Since you have been tagged for the UI change, I'll pass it to you for the UI changes :)

**Comment by k88hudson** [comment]
Thanks nan! I will split off the UI commit in another PR so it's easier for ricky to review :+1:

File: system-addon/content-src/message-center/components/SnippetBase.jsx (hunk 1 above)
Comment (rlr) [reply system-addon/content-src/message-center/components/SnippetBase.jsx]: 👍 as long as you are OK with the color being coupled to the card color.

**Pull request merged**